src/java.base/share/classes/java/util/Properties.java

```diff
@@ -814,10 +821,16 @@ public void save(OutputStream out, String comments)  {
      * character {@code !} then an ASCII {@code #} is written out
      * after that line separator.
      * <p>
-     * Next, a comment line is always written, consisting of an ASCII
-     * {@code #} character, the current date and time (as if produced
-     * by the {@code toString} method of {@code Date} for the
-     * current time), and a line separator as generated by the {@code Writer}.
+     * If the {@systemProperty java.util.Properties.storeDate} is set and
+     * is non-empty (as determined by {@link String#isEmpty()  String.isEmpty}),
+     * a comment line is written as follows.
+     * First, a {@code #} character is written, followed by the contents
+     * of the property, followed by a line separator.
+     * If the system property is not set or is empty, a comment line is written
+     * as follows.
+     * First, a {@code #} character is written, followed by the current date and time
+     * formatted as if by {@link DateTimeFormatter} with the format
+     * {@code "EEE MMM dd HH:mm:ss zzz yyyy"}, followed by a line separator.
      * <p>
      * Then every entry in this {@code Properties} table is
      * written out, one per line. For each entry the key string is
```

Comment:
Sorry for coming late to the party: I don't remember if this was already asked for and rejected - but shouldn't there be a non-normative `@implNote` here to state that the default implementation of this method will write the properties sorted by their keys in alphanumeric ordering?
Otherwise this looks very good!

Reply:
Hello Daniel, you are right - I missed discussing whether or not to include a `@implNote` to explain the natural sorted order of the property keys. When we started this whole PR proposal, Alan had hinted that maybe we should "specify" this behaviour. Should this be a `@implNote` or should this be part of the formal spec like we did for the system property handling?

Reply:
I would leave it as an `@implNote` - or possibly `@implSpec`: depending on whether or not we want all implementations of the spec to behave in this way. However I don't think we would want to prevent subclasses from overriding this behavior and using their own business-logic ordering. So I believe the default behavior should be specified, if only so that subclasses can decide whether or not to override this method, without invalidating what subclasses might currently have implemented - or might wish to implement in the future. The CSR will be a good way to get feedback on whether `@implNote` or `@implSpec` is more appropriate.

Reply:
Done. I've updated the PR to include a `@implNote` specifying the order of the properties. The CSR has been updated too to reflect this.

Reply:
Thanks Jaikiran, - the `@implNote` looks good to me.